src/lib/HST/Frontend/FromGHC.hs

```diff
@@ -0,0 +1,382 @@
+{-# LANGUAGE PackageImports, TypeFamilies #-}
+
+-- | This module contains functions transforming Haskell modules and other
+--   constructs of the AST data structure of @ghc-lib-parser@ into the
+--   corresponding constructs of the AST data structure in the
+--   "HST.Frontend.Syntax" module.
+
+module HST.Frontend.FromGHC where
+
+import           Data.Data                      ( Data )
+import           Data.Maybe                     ( fromMaybe
+                                                , mapMaybe
+                                                )
+
+import qualified "ghc-lib-parser" GHC.Hs       as GHC
+import qualified "ghc-lib-parser" SrcLoc       as GHC
+import qualified "ghc-lib-parser" RdrName      as GHC
+import qualified "ghc-lib-parser" OccName      as GHC
+import qualified "ghc-lib-parser" BasicTypes   as GHC
+import qualified "ghc-lib-parser" Bag          as GHC
+import qualified "ghc-lib-parser" TysWiredIn   as GHC
+import qualified "ghc-lib-parser" Module       as GHC
+import qualified "ghc-lib-parser" Name         as GHC
+import qualified "ghc-lib-parser" TyCon        as GHC
+import qualified "ghc-lib-parser" Type         as GHC
+import qualified "ghc-lib-parser" DynFlags     as GHC
+import qualified "ghc-lib-parser" GHC.Hs.Dump  as GHC
+import qualified "ghc-lib-parser" Outputable   as GHC
+import qualified Language.Haskell.GhclibParserEx.GHC.Settings.Config
+                                               as GHC
+
+import qualified HST.Frontend.Syntax           as S
+
+-------------------------------------------------------------------------------
+-- Type Family Instances                                                     --
+-------------------------------------------------------------------------------
+
+-- | Type representing the AST data structure of @ghc-lib-parser@.
+--
+--   Instantiates the type families for source spans, literals and type
+--   expressions with the concrete types from @ghc-lib-parser@ or wrappers for
+--   these types. Also adds instances for 'S.EqAST' and 'S.ShowAST' to allow
+--   the usage of @==@ and @show@.
+data GHC
+type instance S.SrcSpanType GHC = GHC.SrcSpan
+type instance S.Literal GHC = LitWrapper
+type instance S.TypeExp GHC = TypeWrapper
+
+instance S.EqAST GHC
+instance S.ShowAST GHC
+
+-------------------------------------------------------------------------------
+-- Wrappers for @ghc-lib-parser@ Types                                       --
+-------------------------------------------------------------------------------
+
+-- | Wrapper for the two literal types (for regular and overloaded literals)
+--   used by @ghc-lib-parser@.
+data LitWrapper = Lit (GHC.HsLit GHC.GhcPs)
+                | OverLit (GHC.HsOverLit GHC.GhcPs)
+  deriving Eq
```

Comment:
Shouldn't literals also be compared with `defaultPrintEq` such that source spans are ignored?

Reply:
These literal types don't appear to contain source span information.